### DISCUSSION TITLE
Why the rename?

### DISCUSSION THREAD

**Discussion by davdroman** [pkgxdev/pkgx#763] [Help] [answered]
I don't necessarily hate it. Just curious about it :)

**Comment by mxcl** [comment, marked as answer]
There’s a lot of reasons.

Firstly I do apologize to our early adopters who have faced all the churn in functionality and now a name change, but I really think we have nailed this product now. brand, name and functionality are all on point.

`pkgx` runs anything and provides amazing shell integration that puts the whole open source ecosystem at your fingertips thanks to our trio of powerful commands: `x`, `env` and `dev`. The `x` in `pkgx` is inspired by similar tools like `npx` and `pipx` bringing the power of those ecosystems to a simple command runner. We do the same, but for all open source.

We never were a package manager, I was always trying to create something new. Something that achieved the same goal as package managers but that made using the plethora of amazing tooling that open source provides way more seamless and delightful. Managing packages is lame. **Running Anything** is awesome.

---

tea wasn’t a great name for the tool. There was conflict with gitea and `tee`. It was a meaningless name, much like Homebrew. When I created `brew` I was rebelling against the terrible names that existed then. Open source was full of “purposeful” names that did what they said on the tin. Everything lacked character. There were no Porsche 911s, there were just “automobiles.v8.900hp”. But nowadays every project has a meaningless, abstract name. It’s time for some balance. `pkgx` perfectly describes what it does to anyone familiar with `npx`, and if you’re not familiar, it still strongly suggests what it does.

However tea **is** a great name for the protocol…

---

Finally, over the last 12 months our messaging really struggled because we said we were `brew2` and also a company making an innovative blockchain based protocol that aimed to reward the true value of open source. Trying to merge these things was tricky: the messaging was muddled. People didn’t trust our packager because of crypto. People were confused about our protocol because we were building a package manager.

So we’ve split these parts. pkgx is a core contributor to tea protocol. tea protocol is building a protocol that recognizes open source’s true value using the numerous package manager registries like npm, pypa and crates.io to identify it. One of those registries is `pkgx`.

And with pkgx we will dog food the tea protocol, steeping against our own open source and dependencies and showing the world that at pkgx we truly believe in what we are helping to build with the tea protocol.

**Comment by spencergilbert** [comment]
I'm also curious if the "pantry" will also be renamed to be better aligned with the new branding, it feels a little out of place with `pkgx` vs the original "food" theme.

**Reply by mxcl** [reply]
I considered a rename, but in the end decided it had historical value to us all to keep it. People will always ask, “why is it called the pantry?” and some of us can smile because we know.

Though if you have any great ideas I'm all ears!

**Reply by jhheider** [reply]
`depot`

**Reply by spencergilbert** [reply]
To be clear, no objections or concerns about `pantry` more a curiosity has I happened to started trialing the tool (good timing as it was _during_ the rename, which was a bit of a confusing moment 😅)

**Comment by magnusviri** [comment]
TEA_DIR too.

**Reply by mxcl** [reply]
Is now `PKGX_DIR`

**Comment by DitchComfort** [comment]
👍🏻
